transformer_engine/pytorch/module/base.py

```diff
@@ -1371,6 +1371,7 @@ def get_weight_workspace(
         # Note: Make sure weights have required usages, but do not
         # destroy unnecessary usages since they may be used later.
         if isinstance(tensor, QuantizedTensor):
+            quantizer = tensor._quantizer
```

Comment:
This doesn't really make sense since there's no reason to force the tensor to match its internal quantizer. The purpose of this logic is to force the tensor to match the user-provided quantizer. If the caller has already put in effort to configure a quantizer, we should actually make sure that the output matches that config. The right fix is to change to quantizer config logic to handle the FSDP case.